File: Java/SelectionSort.java
import java.util.*;

public class SelectionSort {
    public static void main(String[] args) throws Exception {
        
        List<Integer> list = new ArrayList<>();
        list.add(5);
        list.add(2);
        list.add(4);
        list.add(6);
        list.add(1);
        list.add(3);
        
        System.out.println(selectionSort(list));
    }
    
    public static List<Integer> selectionSort(List<Integer> list) {
        for (int i = 0; i < list.size(); i++) {
            int minj = i;
            for (int j = i; j < list.size(); j++) {
                if (list.get(minj) > list.get(j)) {
                    int beSortedValue = list.get(j);
                    list.set(j, list.get(minj));                    
                    list.set(minj, beSortedValue);
                }
            }
        }
        return list;
    } 
}

File: Scala/SelectionSort.scala
import scala.collection.mutable.ListBuffer

object SelectionSort extends App{
    
    println(selectionSort(ListBuffer(5, 2, 4, 6, 1, 3)));
    
    /**
     * 選択ソート
     * @param list: リスト
     */
    def selectionSort(list: ListBuffer[Int]): ListBuffer[Int] = {
        for (i <- 0 until list.length) {
            var minj = i;
            for (j <- i until list.length) {
                if (list.apply(minj) > list.apply(j)) {
                    minj = j;
                }
            }
            var beSortedValue = list.apply(i);
            list.update(i, list.apply(minj));
            list.update(minj, beSortedValue);
        }
        return list;
    }
}
